application/src/main/java/org/togetherjava/tjbot/features/javamail/RSSHandlerRoutine.java

```diff
@@ -0,0 +1,412 @@
+     * @return the RSS feed record retrieved from the database
+     */
+    private RssFeedRecord getRssFeedRecordFromDatabase(RSSFeed feedConfig) {
+        return database.read(context -> context.selectFrom(RSS_FEED)
+            .where(RSS_FEED.URL.eq(feedConfig.url()))
+            .limit(1)
+            .fetchAny());
+    }
+
+    /**
+     * Retrieves the last saved date from the database record associated with the given RSS feed
+     * record.
+     * <p>
+     * If the RSS feed record is null, returns an empty {@link Optional}.
+     *
+     * @param rssRecord the RSS feed record to retrieve the last saved date from
+     * @param dateFormatterPattern the pattern used to parse the date from the database record
+     * @return An {@link Optional} containing the last saved date if it could be retrieved and
+     *         parsed successfully, otherwise an empty {@link Optional}
+     */
+    private Optional<ZonedDateTime> getLastSavedDateFromDatabaseRecord(RssFeedRecord rssRecord,
+            String dateFormatterPattern) {
+        if (rssRecord == null) {
+            return Optional.empty();
+        }
+
+        try {
+            ZonedDateTime savedDate =
+                    getZonedDateTime(rssRecord.getLastDate(), dateFormatterPattern);
+            return Optional.of(savedDate);
+        } catch (DateTimeParseException e) {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * Retrieves the latest post date from the given list of items.
+     *
+     * @param items the list of items to retrieve the latest post date from
+     * @param dateFormatterPattern the pattern used to parse the date from the database record
+     * @return the latest post date as a {@link ZonedDateTime} object, or null if the list is empty
+     */
+    private ZonedDateTime getLatestPostDateFromItems(List<Item> items,
+            String dateFormatterPattern) {
+        return items.stream()
+            .map(item -> getDateTimeFromItem(item, dateFormatterPattern))
+            .max(ZonedDateTime::compareTo)
+            .orElse(null);
+    }
+
+    /**
+     * Posts an RSS item to a text channel.
+     *
+     * @param textChannel the text channel to which the item will be posted
+     * @param rssItem the RSS item to post
+     * @param feedConfig the RSS feed configuration
+     */
+    private void postItem(TextChannel textChannel, Item rssItem, RSSFeed feedConfig) {
+        MessageEmbed embed = constructEmbedMessage(rssItem, feedConfig).build();
+        textChannel.sendMessageEmbeds(List.of(embed)).queue();
+    }
+
+    /**
+     * Updates the last posted date to the database for the specified RSS feed configuration.
+     * <p>
+     * This will insert a <b>new</b> entry to the database if the provided {@link RssFeedRecord} is
+     * null.
+     *
+     * @param feedConfig the RSS feed configuration
+     * @param rssFeedRecord the record representing the RSS feed. can be null if not found in the
```

Comment:
"can" should be capitalized (or use a ,)

Reply:
Corrected!